packages/compass-crud/src/stores/crud-store.ts

```diff
@@ -1345,6 +1343,28 @@ class CrudStoreImpl
       countOptions,
     });
 
+    // Only check if index was used if query filter or sort is not empty
+    if (!isEmpty(query.filter) || !isEmpty(query.sort)) {
+      void this.dataService
+        .explainFind(ns, query.filter, findOptions as any, {
+          explainVerbosity: 'queryPlanner',
+          abortSignal: signal,
+        })
+        .then((rawExplainPlan) => {
+          const explainPlan = new ExplainPlan(rawExplainPlan as Stage);
+          this.setState({
+            unindexedQuery: explainPlan.usedIndexes.length === 0,
```

Comment:
I don’t quite think this is the right thing, no?

We don’t want this signal for queries which use 0 indexes, we want it for queries where we’re running a COLLSCAN (even if another stage maybe used an index), right? And, maybe less importantly, we are adding entries to `explainPlan.usedIndexes` for shards even when no index was actually used.

IIUC the compass-explain-plan package uses `explainPlan.isCollectionScan`, and I would do the same here (and potentially even rename `unindexedQuery` to `isCollectionScan` everywhere)

Reply:
Yeah, I guess what I’m saying that whether an index was used _in some way_ isn’t the information that users would actually be looking for :)

Reply:
is it worth factoring this logic in a common place? seems something that may be relevant also for aggregations and the explain panel

Reply:
More common than the `.isCollectionScan` getter?

Reply:
isCollectionScan is already coming from the explain-plan-helper, that's a pretty common place I think 🤔 All the other logic will be different between all three places: findExplain vs findAggregate vs we already have this info in the explain plan modal

Reply:
ohhh yeah i missed half of the conversation :), thanks!